src/main/peripherals/IPeripheral.java

```diff
@@ -0,0 +1,7 @@
+package main.peripherals;
+import java.io.IOException;
+
+public interface IPeripheral {
+	public void performAction(String action) throws ActionNotDefinedException, IOException, InterruptedException;
+	public int getId();
```

Comment:
I doubt if a id stored at the pheripheral is the best idea.
If this is the case it should at least be an abstract class.
Because now we need to add the same line of code to all the implementations and we have no way to force a correct implementation in a later stage (think outside plugins).
Furthermore, if we want a string as an id we have to change it everywhere.